### ISSUE TITLE
GCC12 error for EVUTIL_FALLTHROUGH

### ISSUE THREAD

**Issue by mareksm** [libevent/libevent#1479]
GCC12 (on Debian 12) failing to build the latest/master libevent with the following error:

```
`In file included from src/libevent/include/evthread-internal.h:37,
                 from src/libevent/include/bufferevent-internal.h:38,
                 from src/libevent/bufferevent_filter.c:58:
src/libevent/bufferevent_filter.c: In function 'be_filter_ctrl':
src/libevent/include/util-internal.h:359:28: error: a label can only be part of a statement and a declaration is not a statement [-Wpedantic]
  359 | #define EVUTIL_FALLTHROUGH __attribute__((fallthrough))
      |                            ^~~~~~~~~~~~~
src/libevent/bufferevent_filter.c:620:17: note: in expansion of macro 'EVUTIL_FALLTHROUGH'
  620 |                 EVUTIL_FALLTHROUGH;
      |                 ^~~~~~~~~~~~~~~~~~
`
```

The C language standard only allows statements​ to follow a label. The language does not group declarations in the same category as statements.

```
gcc --version
gcc (Debian 12.2.0-14) 12.2.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by mareksm** [comment]
The fix is simple:

`

diff --git a/server/src/libevent/bufferevent_filter.c b/server/src/libevent/bufferevent_filter.c
index a7bdedd..00d1258 100644
--- a/server/src/libevent/bufferevent_filter.c
+++ b/server/src/libevent/bufferevent_filter.c
@@ -616,7 +616,9 @@ be_filter_ctrl(struct bufferevent *bev, enum bufferevent_ctrl_op op,
                EVUTIL_FALLTHROUGH;
 
        case BEV_CTRL_CANCEL_ALL:
+       {
                EVUTIL_FALLTHROUGH;
+       }
        default:
                return -1;
        }

`

**Comment by azat** [comment]
Thanks for the report. Can you submit a PR with this patch?

**Comment by azat** [comment]
Fixed in #1481
